src/Elasticsearch.Net/Connection/HttpConnection.cs

```diff
@@ -197,73 +179,63 @@ private void SetBasicAuthenticationIfNeeded(Uri uri, HttpWebRequest request, IRe
 				userInfo = requestSpecificConfig.BasicAuthorizationCredentials.ToString();
 
 			if (!userInfo.IsNullOrEmpty())
-				request.Headers["Authorization"] = "Basic " + Convert.ToBase64String(Encoding.UTF8.GetBytes(userInfo));
+				request.Headers.Authorization = AuthenticationHeaderValue.Parse("Basic " + Convert.ToBase64String(Encoding.UTF8.GetBytes(userInfo)));
 		}
 
-		protected virtual HttpWebRequest CreateWebRequest(Uri uri, string method, byte[] data, IRequestConfiguration requestSpecificConfig)
+		protected virtual HttpRequestMessage CreateHttpRequestMessage(Uri uri, string method, byte[] data, IRequestConfiguration requestSpecificConfig)
 		{
-			var request = (HttpWebRequest)WebRequest.Create(uri);
-			request.Accept = "application/json";
-			request.ContentType = "application/json";
-			request.Pipelined = this.ConnectionSettings.HttpPipeliningEnabled
-				|| (requestSpecificConfig != null && requestSpecificConfig.EnableHttpPipelining);
-
-			if (this.ConnectionSettings.EnableCompressedResponses
-				|| this.ConnectionSettings.EnableHttpCompression)
+			var content = data == null ? null : new ByteArrayContent(data)
 			{
-				request.AutomaticDecompression = DecompressionMethods.GZip | DecompressionMethods.Deflate;
-				request.Headers.Add("Accept-Encoding", "gzip,deflate");
-				if (this.ConnectionSettings.EnableHttpCompression)
-					request.Headers.Add("Content-Encoding", "gzip");
-			}
+				Headers =
+				{
+					ContentType = MediaTypeHeaderValue.Parse("application/json")
+				}
+			};
+			var request = new HttpRequestMessage
+			{
+				RequestUri = uri,
+				Method = new HttpMethod(method),
+				Content = content,
+				Headers =
+				{
+					Accept =
+					{
+						MediaTypeWithQualityHeaderValue.Parse("application/json")
+					}
+				}
+			};

```

Comment:
It seems like Microsoft has made changes to the HttpRequestMessage and that there is a difference between the one we use in net framework, and the one we will use in the future on .NET 6. We should therefore make sure that we investigate this and other things when upgrading.

No impact on this PR however.

Reply:
I agree, and also that we should have a task for upgrading these libraries to .NET 6 when the rest is migrated.